scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala

```diff
@@ -32,9 +32,16 @@ case object RedundantBraces extends Rewrite {
     def isIdentifierAtStart(value: String) =
       value.nonEmpty && (Character.isLetterOrDigit(value.head) || value.head == '_')
 
+    // hack for literal identifiers detection (`type`, `val`, ...)
+    def isLiteralIdentifier(arg: Term.Name): Boolean =
+      (arg.pos.end - arg.pos.start) - arg.value.length() == 2
+
+    def shouldTermBeEscaped(arg: Term.Name): Boolean =
```

Comment:
Could we add a comment that refers to https://github.com/scalameta/scalafmt/issues/1420 ?
This solution is a bit hacky and worth explaining why this condition is required :)

Reply:
done